purchase_discount/views/product_supplierinfo_view.xml

```diff
@@ -0,0 +1,23 @@
+<odoo>
+
+    <record model="ir.ui.view" id="product_supplierinfo_form_view">
+         <field name="model">product.supplierinfo</field>
+         <field name="inherit_id" ref="product.product_supplierinfo_form_view"/>
+         <field name="arch" type="xml">
+              <field name="price" position="after">
+                  <field name="discount"/>
+              </field>
+         </field>
+     </record>
+
+    <record model="ir.ui.view" id="product_supplierinfo_tree_view">
+         <field name="model">product.supplierinfo</field>
+         <field name="inherit_id" ref="product.product_supplierinfo_tree_view"/>
+         <field name="arch" type="xml">
+              <field name="price" position="after">
```

Comment:
Discount field in product supplier info form isn't properly shown. You should use instead

```suggestion
              <label for="date_start" position="before">
                   <label for="discount"/>
                   <div>
                      <field name="discount" class="oe_inline"/>
                  </div>              
              </label>
```

Reply:
Done!!!